pkg/kubelet/server/server.go

```diff
@@ -240,11 +253,27 @@ func NewServer(
 	resourceAnalyzer stats.ResourceAnalyzer,
 	auth AuthInterface,
 	kubeCfg *kubeletconfiginternal.KubeletConfiguration) Server {
+	var tp oteltrace.TracerProvider
+	if utilfeature.DefaultFeatureGate.Enabled(features.KubeletTracing) {
+		opts := []otlpgrpc.Option{}
+		if kubeCfg.Tracing.Endpoint == nil {
+			opts = append(opts, otlpgrpc.WithEndpoint(*kubeCfg.Tracing.Endpoint))
+		}
+		opts = append(opts, otlpgrpc.WithInsecure())
+		var err error
+		tp, err = traces.InitTracing(context.Background(), host.GetHostname(), kubeletServiceName, kubeCfg.Tracing.SamplingRatePerMillion, opts)
+		if err != nil {
+			klog.ErrorS(err, "Failed to configure kubelet traces")
+			os.Exit(1)
+		}
+		otel.SetTracerProvider(tp)
```

Comment:
If possible, we should avoid using the global tracer provider and propagators.  For each client/server, we should pass propagators and tracer provider explicitly.

Reply:
done, removed this.